### ISSUE TITLE
Committing to seperate branches

### ISSUE THREAD

**Issue by rioforce** [BoatsAreRockable/animator#66]
This isn't really an issue, but I think it needs to be brought up. You, Boats, can push commits directly to the master branch. Contributors working on forks cannot commit to the master. We have to submit pull requests. In my opinion (and I think this is actually a good practice of using Git), you should create branches, edit on those branches and submit pull requests. Then you can push them to the main. This way, contributors have a bit of time to see the commits you are making without having to change their code every time you make a commit, and they can make their changes uniform with yours a bit later, before they make a pull request. This also permits code and design review to see if everything works and acts well (especially review by other contributors).

In short, the master should be a cleaner, more stable branch, and branches are unstable WIP things. Of course, some small things can still be merged into master without pull requests, but large things like custom scrollbars, or custom resolution, should be made on a separate branch and submitted with a pull request.

It just helps everyone around. :wink:

**Comment by BoatsAreRockable** [comment]
Thanks for informing me! I had a feeling the way I was doing it wasn't quite right.

I shall aim to do this from now on.

**Comment by BoatsAreRockable** [comment]
Question: When I submit pull releases that are pretty stable, should I wait for feedback / approval first or is alright for me to merge them straight away?

**Comment by rioforce** [comment]
You don't have to wait if it's stable. However, major changes are best when set for a little while (anywhere from 2 hours to 2 days, depending if you need feedback or if we're active, etc). Basically, it's gut instinct. ;)
